ui/dashboards/src/components/GridLayout/GridLayout.tsx

```diff
@@ -37,6 +39,17 @@ export function GridLayout(props: GridLayoutProps) {
   const [isOpen, setIsOpen] = useState(!groupDefinition.isCollapsed ?? true);
   const { isEditMode } = useEditMode();
 
+  const handleLayoutChange = (currentLayout: Layout[], allLayouts: Layouts) => {
+    // Using the value from `allLayouts` instead of `currentLayout` because of
+    // a bug in react-layout-grid where `currentLayout` does not adjust properly
+    // when going to a smaller breakpoint and then back to a larger breakpoint.
+    // https://github.com/react-grid-layout/react-grid-layout/issues/1663
+    const smallLayout = allLayouts[STORED_LAYOUT_BREAKPOINT];
+    if (smallLayout) {
+      updatePanelGroupLayouts(smallLayout);
```

Comment:
I _think_ always ensuring we store the `sm` breakpoint should also avoid weirdness where someone tries to edit and save their dashboard on an extra small screen size.

Reply:
Not a big deal, but you could maybe called `smallLayout` => `storedLayout`, or you could call `STORED_LAYOUT_BREAKPOINT` => `SMALL_LAYOUT_BREAKPOINT`

Just calling out because if `STORED_LAYOUT_BREAKPOINT` value did change for some reason, it would no longer be the `smallLayout`